### PR TITLE
Support power operator

### LINKED ISSUE #253: Support power operation **

### PR DESCRIPTION
**Pull request by DJRHails**
Fixes #253

### CODE REVIEW

File: Sources/AST/Token.swift [hunk 1]
```diff
@@ -94,6 +95,7 @@ extension Token {
         case .minus, .overflowingMinus: return 20
         case .times, .overflowingTimes: return 30
         case .divide: return 30
+        case .power: return 35
         case .ampersand: return 35
```
Comment (bearbin) [inline Sources/AST/Token.swift]: It looks like the precendence here should have been incremented, so as to keep the increasing sequence.

File: Sources/IRGen/Runtime/IULIARuntimeFunction.swift [hunk 2]
```diff
@@ -327,5 +333,16 @@ struct IULIARuntimeFunctionDeclaration {
...
-}
  
+  //TODO: Add checks for invalid power calls
+  static let power =
+  """
+  function flint$power(b, e) -> ret {
+    ret := 1
+    for { let i := 0 } lt(i, e) { i := add(i, 1) }
+    {
+        ret := mul(ret, b)
```
Comment (bearbin) [inline Sources/IRGen/Runtime/IULIARuntimeFunction.swift]: Perhaps this should use the `flint$mul` non-overflowing multiplication function?

File: Sources/AST/Token.swift (hunk 1 above)
Comment (DJRHails) [reply Sources/AST/Token.swift]: As in make it 31?

Comment (bearbin) [reply Sources/AST/Token.swift]: That, or make the ones following 40, 45, etc.

**Comment by franklinsch** [comment]
Add tests for checking overflows

**Pull request merged**